Gordon360/Controllers/RecIM/ActivitiesController.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Services.RecIM;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    public class ActivitiesController : GordonControllerBase
+    {
+        private readonly IActivityService _activityService;
+
+        public ActivitiesController(IActivityService activityService)
+        {
+            _activityService = activityService;
+        }
+
+        ///<summary>Gets a list of all Activities</summary>
+        ///<param name="active"> Optional active parameter </param>
+        /// <returns>
+        /// All Existing Activities 
+        /// </returns>
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<Activity>> GetActivities(Boolean active = false)
+        {
+            if (active)
+            {
+                var result = _activityService.GetActivitiesByTime(null);
+                return Ok(result);
+            } else
+            {
+                var result = _activityService.GetActivities();
+                return Ok(result);
+            }
+        }
+
+        ///<summary>Gets a list of all Activities after given time</summary>
+        /// <returns>
+        /// All Existing Activities 
+        /// </returns>
+        [HttpGet]
+        [Route("?date={time}")]
+        public ActionResult<IEnumerable<Activity>> GetAllActivities(DateTime time)
```

Comment:
FYI, this is not the preferred way to do query string binding in ASP.NET Core. Use the `FromQuery` attribute instead:
```suggestion
        [Route("")]
        public ActionResult<IEnumerable<Activity>> GetAllActivities([FromQuery] DateTime time)
```